expression/builtin_miscellaneous_test.go

```diff
@@ -295,8 +299,12 @@ func TestInet6AtoN(t *testing.T) {
 		f, err := fc.getFunction(ctx, datumsToConstants([]types.Datum{ip}))
 		require.NoError(t, err)
 		result, err := evalBuiltinFunc(f, chunk.Row{})
-		require.NoError(t, err)
-		testutil.DatumEqual(t, types.NewDatum(test.expect), result)
+		expect := types.NewDatum(test.expect)
+		if expect.IsNull() {
+			result.SetNull()
+			require.True(t, terror.ErrorEqual(err, errWrongValueForType))
+		}
+		testutil.DatumEqual(t, expect, result)
```

Comment:
I suggest you to test like this:

```
if expect.IsNull() {
			require.True(t, terror.ErrorEqual(err, errWrongValueForType))
} else {
            require.NoError(t, err)
		    testutil.DatumEqual(t, expect, result)
}
```

Reply:
OK. A bit doubt in bench_test.go, with this vec bench test it seems the original test cases generate IP strings use rand()  machnism, not set expect result as above, thats why I did not change it the same way as we agreed already. I thought what we want to test already proved in above case, so in vec bench case may just avoid show error is ok to reach the vec bench test aim also, dont need to test the above logic again. is that reasonable?

Reply:
OK. A bit doubt in bench_test.go, with this vec bench test it seems the original test cases generate IP strings use rand() machnism, not set expect result as above, thats why I did not change it the same way as we agreed already. I thought what we want to test already proved in above case, so in vec bench case may just avoid show error is ok to reach the vec bench test aim also, dont need to test the above logic again. is that reasonable? anyway, I will deep to learn the logic to find. THKS

Reply:
That makes sense. I'll double check if your logic is correct.

Reply:
But better add `NoError`, asserting more is better than nothing.

Reply:
> OK. A bit doubt in bench_test.go, with this vec bench test it seems the original test cases generate IP strings use rand() machnism, not set expect result as above, thats why I did not change it the same way as we agreed already. I thought what we want to test already proved in above case, so in vec bench case may just avoid show error is ok to reach the vec bench test aim also, dont need to test the above logic again. is that reasonable? anyway, I will deep to learn the logic to find. THKS

A better fix would be removing invalid cases here https://github.com/pingcap/tidb/blob/f9a324829b71a24141d5326ddc0026f6b12125c3/expression/builtin_miscellaneous_vec_test.go#L51-L66. And all cases will pass, then there is no need to use the result as expectation.

Since your midifications applies to a lot, a lot of tests rather than INET_ATON cases. So no, your changes may let other cases fail without even knowing.

Honestly, just don't use the result as expectation. As I said, it breaks the fundamental rules of testing. Tests should be reliable and sound forever. Tests are to fail, not to pass all the time.

Reply:
BTW, resolve the comment, after you address the corresponding changes, like what I did :)

Reply:
THKS! Did you mean just leave the test errors there with bench_test.go because not to pass all the time?

Reply:
> THKS! Did you mean just leave the test errors there with bench_test.go because not to pass all the time?

I mean fix it by removing tests that will fail, that is recover the test code to the original.

Since failed tests are covered in other tests, it is ok.

Reply:
I just did new fix, follow the above logic. Actually before I tried this, but because the bench_test inet_aton also generate IP string by rand way, so err always happens, and I dont want to change too much of the previous logic before. Now I think it make no senso for Inet_aton, so just remove those random string gen. please check and THKS.

Reply:
The last thing, resolve the conversation/comment after you addressing the problem.